cron/src/main/java/com/foursquare/service/impl/TrendingCategoriesServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.foursquare.service.impl;
+
+import com.foursquare.entity.Category;
+import com.foursquare.entity.User;
+import com.foursquare.entity.Venue;
+import com.foursquare.repository.UserRepository;
+import com.foursquare.repository.VenueRepository;
+import com.foursquare.service.TrendingCategoriesService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class TrendingCategoriesServiceImpl implements TrendingCategoriesService {
+
+    @Autowired
+    private VenueRepository venueRepository;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private JavaMailSender mailSender;
+
+    public void emailTrendingCategories() {
+        Calendar calendar = Calendar.getInstance();
+
+        if(calendar.get(Calendar.MONTH) == 0) {
+            calendar.set(Calendar.MONTH, 11);
+            calendar.set(Calendar.YEAR, calendar.get(Calendar.YEAR) - 1);
+        } else {
+            calendar.set(Calendar.MONTH, calendar.get(Calendar.MONTH) - 1);
+        }
+
+        List<Venue> allMonthlyVenues = venueRepository.findAllByAddedAtGreaterThanEqual(calendar.getTime());
+
+        if (!allMonthlyVenues.isEmpty()) {
+
+            HashMap<Category, Integer> monthlyCategoriesOfVenues = new HashMap<>();
+
+            for (Venue venue : allMonthlyVenues) {
+                for (Category category : venue.getCategories()) {
+                    if (!monthlyCategoriesOfVenues.containsKey(category)) {
+                        monthlyCategoriesOfVenues.put(category, 0);
+                    }
+                }
+            }
+
+            allMonthlyVenues.stream().forEach(venue ->
+                    venue.getCategories().stream().forEach(category ->
+                            monthlyCategoriesOfVenues.put(category, monthlyCategoriesOfVenues.get(category) + 1)));
+
+            List<Map.Entry<Category, Integer>> sortedMonthlyCategoriesByValue = monthlyCategoriesOfVenues.entrySet().stream()
+                    .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                    .collect(Collectors.toList());
+
+            mailing(getTrendingCategoriesFromSortedList(sortedMonthlyCategoriesByValue));
+        }
+    }
+
+    private List<Map.Entry<Category, Integer>> getTrendingCategoriesFromSortedList(List<Map.Entry<Category, Integer>> sortedCategoriesList) {
```

Comment:
1) This method is unreadable.
2) Seems like this method won't work properly for list size 2
Please use top 3 only. Even if there are same number of usages for more than three categories.